src/ssp.cpp

```diff
@@ -204,30 +118,126 @@ void InitQuad(SSP_INIT_ARGS)
     ssp->Nz = ssp->NPts;
 }
 
-void InitializeSSP(SSP_INIT_ARGS)
+/**
+* Update the SSP parameters. Safe to call multiple times with flags.
+* Be sure to flag ssp->dirty if you change the SSP externally.
+* TODO: You have to provide FileRoot if you use a 2D SSP.
+*/
+void UpdateSSP(vec2 x, real Depth, real freq, const real& fT, SSPStructure* ssp, 
+    PrintFileEmu& PRTFile, const AttenInfo* atten, std::string FileRoot)
 {
-    switch(ssp->Type){
-    case 'N': // N2-linear profile option
-        Initn2Linear(SSP_CALL_INIT_ARGS); break;
-    case 'C': // C-linear profile option
-        InitcLinear (SSP_CALL_INIT_ARGS); break;
-    case 'P': // monotone PCHIP ACS profile option
-        InitcPCHIP  (SSP_CALL_INIT_ARGS); break;
-    case 'S': // Cubic spline profile option
-        InitcCubic  (SSP_CALL_INIT_ARGS); break;
-    case 'Q':
-        InitQuad    (SSP_CALL_INIT_ARGS); break;
-    /* case 'H':
-        // this is called by BELLHOP3D only once, during READIN
-        // possibly the logic should be changed to call EvaluateSSP2D or 3D
-        x3 = vec3(RL(0.0), RL(0.0), x.y);
-        InitHexahedral(x3, freq, ssp); break; */
-    case 'A': // Analytic profile option
-        break; //LP: No init for analytic.
-    default:
-        GlobalLog("InitializeSSP: Invalid profile option %c\n", ssp->Type);
+    if (ssp->dirty) {
```

Comment:
Probably would be cleaner as `if(!ssp->dirty) return;` and then no need for braces for the rest of the function.